homeassistant/components/sensor/elkm1.py

```diff
@@ -0,0 +1,315 @@
+
+        attrs = self.initial_attrs()
+        attrs['area'] = self._element.area + 1
+        attrs['temperature'] = self._element.temperature
+        attrs['last_user_time'] = self._element.last_user_time.isoformat()
+        attrs['last_user'] = self._element.last_user + 1
+        attrs['code'] = self._element.code
+        attrs['last_user_name'] = username(self._elk, self._element.last_user)
+        return attrs
+
+    def _element_changed(self, element, changeset):
+        self._state = temperature_to_state(self._element.temperature, -40)
+
+    async def async_added_to_hass(self):
+        """Register callback for ElkM1 changes and update entity state."""
+        await super().async_added_to_hass()
+        self.hass.data[ELK_DOMAIN]['keypads'][
+            self._element.index] = self.entity_id
+
+
+class ElkPanel(ElkSensor):
+    """Representation of an Elk-M1 Panel."""
+
+    async def async_added_to_hass(self):
+        """Register callback for ElkM1 changes."""
+        await super().async_added_to_hass()
+        async_dispatcher_connect(
+            self.hass, '{}_{}'.format(SIGNAL_SPEAK_WORD, self.entity_id),
+            self._speak_word)
+        async_dispatcher_connect(
+            self.hass, '{}_{}'.format(SIGNAL_SPEAK_PHRASE, self.entity_id),
+            self._speak_phrase)
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend."""
+        return "mdi:home"
+
+    @property
+    def device_state_attributes(self):
+        """Attributes of the sensor."""
+        attrs = self.initial_attrs()
+        attrs['elkm1_version'] = self._element.elkm1_version
+        attrs['system_trouble_status'] = self._element.system_trouble_status
+        attrs['xep_version'] = self._element.xep_version
+        return attrs
+
+    def _element_changed(self, element, changeset):
+        if self._elk.is_connected():
+            self._state = 'Paused' if self._element.remote_programming_status \
+                else 'Connected'
+        else:
+            self._state = 'Disconnected'
+
+    async def _speak_word(self, number):
+        """Speak a word on the panel."""
+        self._element.speak_word(number)
+
+    async def _speak_phrase(self, number):
+        """Speak a phrase on the panel."""
+        self._element.speak_phrase(number)
+
+
+class ElkSetting(ElkSensor):
+    """Representation of an Elk-M1 Setting."""
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend."""
+        return 'mdi:numeric'
+
+    def _element_changed(self, element, changeset):
+        self._state = self._element.value
+
+    @property
+    def device_state_attributes(self):
+        """Attributes of the sensor."""
+        from elkm1_lib.const import SettingFormat
+        attrs = self.initial_attrs()
+        attrs['value_format'] = SettingFormat(
+            self._element.value_format).name.lower()
+        return attrs
+
+
+class ElkThermostat(ElkSensor):
```

Comment:
Is this a pure read only integration, no temperature control? It should be a sensor and not a climate entity?

Reply:
I'm sure I read, but cannot find right now, that each of the sensors of a device should be represented as separate entities. Thus this thermostat entity. Perhaps @BioSehnsucht can comment as I'm guessing he read the same thing from which code followed.

The next PR is for `climate` in which case the temperature will be available from that. Delete?

Reply:
If this thermostat reading is the same device feature that will go in the climate entity, yes then remove this sensor type.

Reply:
TL;DR : I think it's fine to remove the ElkThermostat sensors for now. When we get Omni protocol re-implemented to support the additional sensors, we can add those additional sensors back in.

At one time I had implemented the necessary protocol-within-a-protocol to be able to gather additional temperature sensors from Elk supported climate devices (specifically HAI Omnistat devices) which could have up to 4 temperature sensors in total (one of which was the thermostat itself). However at the moment we haven't re-implemented that again after switching from my library to @gwww 's so thermostat sensors are redundant, and once we re-implement the protocol inception necessary to support it, we can revisit and add just the 2nd-4th temp sensors as individual sensors. (In actuality, at the time, the thermostat sensor that was part of the climate entity was duplicated as a sensor because reasons ... so I always had duplicated sensors for climate)